crates/js-component-bindgen/src/ts_bindgen.rs

```diff
@@ -495,139 +543,135 @@ impl TsBindgen {
+            let is_world_export = false; // TODO
+            let async_funcs = if is_world_export {
+                self.async_exports.clone()
+            } else {
+                self.async_imports.clone()
+            };
+            let iface = &resolve.interfaces[id];
+            let package = resolve
+                .packages
+                .get(iface.package.expect("missing package on interface"))
+                .expect("unexpectedly missing package");
+            for (_, func) in iface.functions.iter() {
+                // Ensure that the function  the world item for stability guarantees and exclude if they do not match
+                if !feature_gate_allowed(resolve, package, &func.stability, &func.name)
+                    .expect("failed to check feature gate for function")
+                {
+                    continue;
                 }
+                let func_name = &func.name;
+                let is_async = is_world_export && async_funcs.contains(func_name)
+                    || async_funcs.contains(&format!("{id_name}#{func_name}"))
+                    || id_name
+                        .find('@')
+                        .map(|i| {
+                            async_funcs
+                                .contains(&format!("{}#{func_name}", id_name.get(0..i).unwrap()))
+                        })
+                        .unwrap_or(false);
+                gen.ts_func(func, false, true, is_async);
             }
-        }
-
-        uwriteln!(gen.src, "}}");
-
-        gen.types(id);
-        gen.post_types();
 
-        let src = gen.finish();
+            gen.types(id);
+            gen.post_types();
 
-        files.push(file_name, src.as_bytes());
+            let (src, references) = gen.finish();
+            files.push(&file_name, generate_references(&references).as_bytes());
+            files.push(&file_name, src.as_bytes());
+        }
 
-        local_name
+        if self.is_guest {
+            self.references.insert(format!("./{file_name}"));
+            id_name
+        } else {
+            format!("./{file_stem}.js")
+        }
     }
 }
 
 impl<'a> TsInterface<'a> {
-    fn new(resolve: &'a Resolve, is_root: bool) -> Self {
+    fn new(resolve: &'a Resolve, is_root: bool, is_guest: bool) -> Self {
         TsInterface {
             is_root,
+            is_guest,
             src: Source::default(),
             resources: BTreeMap::new(),
             local_names: LocalNames::default(),
             resolve,
             has_constructor: false,
             needs_ty_option: false,
             needs_ty_result: false,
+            needs_module_end: false,
+            references: Default::default(),
+        }
+    }
+
+    fn begin(&mut self, id_name: &str) {
+        if self.is_guest {
+            uwriteln!(self.src, "declare module '{id_name}' {{");
+            self.needs_module_end = true;
+        } else {
+            uwriteln!(self.src, "// interface {id_name}");
```

Comment:
This could be a good place for the `@module` doc comment which would then be included in type hints on all exports:

```suggestion
            uwriteln!(self.src, "/** @module Interface {id_name} **/");
```